### ISSUE TITLE
Implement start via intent

### ISSUE THREAD

**Issue by emanuele-f** [emanuele-f/PCAPdroid#33]
The ability to control the capture via intents would enable other apps to use PCAPdroid as a capture tool. The requirements are:

- start and stop the app, with ability to specify the configuration (e.g. dump mode and filter)
- avoid opening the pcapdroid gui
- ask for user confirmation the first time another app tries to control Pcapdroid to prevent malicious apps from dumping users traffic

02f238a7b2c79b3a6c175cee6e063e9c32a237ba implemented the ability to start PCAPdroid via an intent, for example:

```
adb shell am start -e autostart true -e proxyIp 192.168.200.2 -e proxyPort 1234 -e collectorIp 192.168.200.3 -e collectorPort 8080 -e filterPackage com.google.ar.core -n com.emanuelef.remote_capture/.MainActivity
```

However, there are the following issues which prevent the functionality from being functional and secure:

- Since permissions are bypassed (normally an app would require the BIND_VPN_SERVICE permission to start the CaptureService), a malicious app could start the PCAPdroid app to mirror the device traffic to a remote server via the UDP exporter (since it can change the app configuration).
- Rotating the device/resuming the app (thus calling onCreate again) restarts the app as the getIntent still returns the autostart intent.
- Sending a new intent does not restart the app if it's in foreground/background (OnNewIntent is not called so I've removed it)
- Minimal docs on how to use this would be useful (to https://github.com/emanuele-f/PCAPdroid/tree/gh-pages, after these issues are fixed)

Update: due to the issues above, the commit is now reverted.

**Comment by emanuele-f** [comment]
The [latest debug APK](https://github.com/emanuele-f/PCAPdroid/blob/649c131239b92cce1e210a62c98ecc2d183ec5fe/PCAPdroid-debug-c497a95.apk?raw=true) implements the ability to control the packet capture via Intents. Check out https://github.com/emanuele-f/PCAPdroid/blob/master/docs/app_api.md for documentation and examples. Any feedback on this is very appreciated.

**Comment by emanuele-f** [comment]
This is now available in the [1.4.2 release](https://github.com/emanuele-f/PCAPdroid/releases/tag/v1.4.2)
